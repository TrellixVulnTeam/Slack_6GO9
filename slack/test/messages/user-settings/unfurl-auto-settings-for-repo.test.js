const { UnfurlAutoSettingsForRepo } = require('../../../lib/messages/user-settings');

describe('UnfurlAutoSettingsForRepo rendering', () => {
  test('works when repo is unfurled in all channels', async () => {
    const actionValue = '12345|integrations/snappydoo';
    const slackUser = {
      settings: {
        unfurlPrivateResources: {
          12345: ['all', 'C1234'],
        },
      },
    };
    const message = new UnfurlAutoSettingsForRepo(actionValue, slackUser);
    expect(message.getAttachment()).toMatchSnapshot();
  });

  test('works when repo is unfurled only in some channels', async () => {
    const actionValue = '12345|integrations/snappydoo';
    const slackUser = {
      settings: {
        unfurlPrivateResources: {
          12345: ['C4321', 'C1234'],
        },
      },
    };
    const message = new UnfurlAutoSettingsForRepo(actionValue, slackUser);
    expect(message.getAttachment()).toMatchSnapshot();
  });
});
