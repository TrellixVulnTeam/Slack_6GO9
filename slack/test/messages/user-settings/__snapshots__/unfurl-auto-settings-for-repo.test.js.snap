// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnfurlAutoSettingsForRepo rendering works when repo is unfurled in all channels 1`] = `
Object {
  "actions": Array [
    Object {
      "confirm": Object {
        "dismiss_text": "Cancel",
        "ok_text": "Yes",
        "text": "This will disable automatic rich previews for links you post linking to \`integrations/snappydoo\` in *all* channels",
        "title": "Are you sure?",
      },
      "name": "remove-repo",
      "style": "danger",
      "text": "Disable automatic rich previews",
      "type": "button",
      "value": "12345|integrations/snappydoo",
    },
  ],
  "callback_id": "unfurl-settings-auto",
  "color": "#24292f",
  "mrkdwn_in": Array [
    "text",
  ],
  "text": "Links you post linking to \`integrations/snappydoo\` automatically get a rich preview in all channels.",
  "title": "Automatic rich previews",
}
`;

exports[`UnfurlAutoSettingsForRepo rendering works when repo is unfurled only in some channels 1`] = `
Object {
  "actions": Array [
    Object {
      "confirm": Object {
        "dismiss_text": "Cancel",
        "ok_text": "Yes",
        "text": "This will disable automatic rich previews for links you post linking to \`integrations/snappydoo\` in *all* channels",
        "title": "Are you sure?",
      },
      "name": "remove-repo",
      "style": "danger",
      "text": "Disable automatic rich previews",
      "type": "button",
      "value": "12345|integrations/snappydoo",
    },
  ],
  "callback_id": "unfurl-settings-auto",
  "color": "#24292f",
  "mrkdwn_in": Array [
    "text",
  ],
  "text": "Links you post linking to \`integrations/snappydoo\` automatically get a rich preview in the following channels:
<#C4321>,
<#C1234>",
  "title": "Automatic rich previews",
}
`;
